go/host/storage/hostdb/batch.go

```diff
@@ -37,6 +38,9 @@ func AddBatch(dbtx *dbTransaction, statements *SQLStatements, batch *common.ExtB
 		extBatch,                     // ext_batch
 	)
 	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "unique") {
```

Comment:
this fixes the bug that returned the wrong type of error when it failed with the constraint bug